pontoon/sync/core.py

```diff
@@ -20,6 +21,27 @@
 
 log = logging.getLogger(__name__)
 
+def run_only_one(func):
+    """
+    Decorator ensures that there's only one running task with given task_name.
+    """
+
+    task_name = "celery.{}".format(func.__name__)
+    def wrapped_func(*args, **kwargs):
+        # Moved import here to mock cache during tests.
+        from django.core.cache import cache
+
+        # Acquire the lock
+        if not cache.add(task_name, True):
```

Comment:
What do you mean? If the timeout is three hours and a sync job exits without releasing the lock, then 3 hours later the lock will expire and the next sync job will run again, and either succeed or fail normally. Why would this cause an issue?